cmd/es-rollover/app/actions.go

```diff
@@ -37,28 +39,38 @@ type Action interface {
 
 // ActionExecuteOptions are the options passed to the execute action function
 type ActionExecuteOptions struct {
-	Args     []string
-	Viper    *viper.Viper
-	Logger   *zap.Logger
-	TLSFlags tlscfg.ClientFlagsConfig
+	Args      []string
+	Viper     *viper.Viper
+	Logger    *zap.Logger
+	TLSConfig configtls.ClientConfig
 }
 
 // ActionCreatorFunction type is the function type in charge of create the action to be executed
 type ActionCreatorFunction func(client.Client, Config) Action
 
+func getTLSConfig(tlsConfig configtls.ClientConfig, logger *zap.Logger) (*tls.Config, error) {
```

Comment:
I don't believe you need this helper. See below for OTEL's implementation of `LoadTLSConfig`
```go
// LoadTLSConfig loads the TLS configuration.
func (c ClientConfig) LoadTLSConfig(_ context.Context) (*tls.Config, error) {
	if c.Insecure && !c.hasCA() {
		return nil, nil
	}

	tlsCfg, err := c.loadTLSConfig()
	if err != nil {
		return nil, fmt.Errorf("failed to load TLS config: %w", err)
	}
	tlsCfg.ServerName = c.ServerName
	tlsCfg.InsecureSkipVerify = c.InsecureSkipVerify
	return tlsCfg, nil
}
As you can see, it mostly performs the same validations that this function is performing so calling it directly should be sufficient. 
```